source/distributed/tria_base.cc

```diff
@@ -893,6 +893,7 @@ namespace parallel
           const auto &cell_status = std::get<0>(*cell_rel_it);
           const auto &dealii_cell = std::get<1>(*cell_rel_it);
 
+#ifdef DEBUG
```

Comment:
There is nothing left from the following `switch` statement if you remove the assertions, and so I'd rather not add these `#ifdef DEBUG` markers. The compiler will be able to remove the whole thing.